src/main/java/org/example/domain/User.java

```diff
@@ -0,0 +1,60 @@
+package org.example.domain;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Version;
+
+@Entity
+public class User {
+
+  @Id
+  private UUID id;
+
+  @Column
+  private String name;
+
+  @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
fwiw removing `orphanRemoval = true` results in all 3 roles being persisted to the db